python/src/indices.rs

```diff
@@ -269,12 +287,97 @@ pub fn shuffle_transformed_vectors(
     }
 }
 
+async fn do_load_shuffled_vectors(
+    filenames: Vec<String>,
+    dir_path: &str,
+    dataset: &Dataset,
+    column: &str,
+    mut ivf_model: IvfModel,
+    pq_model: ProductQuantizer,
+) -> PyResult<()> {
+    let (_, path) = object_store_from_uri_or_path(dir_path).await?;
+    let streams = load_partitioned_shuffles(path.clone(), filenames)
+        .await
+        .infer_error()?;
+
+    let obj_store = dataset.ds.object_store();
+    let path = dataset.ds.indices_dir().child("shuffled_vectors.idx");
```

Comment:
Again, we want to mirror existing practices.  The index should be written to:

```
let path = dataset.ds.indices_dir().child(some_uuid).child("index.idx");
```

Where `some_uuid` is a v4 UUID.  The uuid will then need to be passed into `commit_existing_index`.